test.disabled/ejabberd_tests/tests/push_SUITE.erl

```diff
@@ -0,0 +1,715 @@
+
+%% ----------------------------------
+%% Stanzas
+%% ----------------------------------
+
+disable_stanza(JID, undefined) ->
+    disable_stanza([
+        {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+        {<<"jid">>, JID}
+    ]);
+disable_stanza(JID, Node) ->
+    disable_stanza([
+        {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+        {<<"jid">>, JID},
+        {<<"node">>, Node}
+    ]).
+disable_stanza(JID) when is_binary(JID) ->
+    disable_stanza(JID, undefined);
+disable_stanza(Attrs) when is_list(Attrs) ->
+    escalus_stanza:iq(<<"set">>, [#xmlel{name = <<"disable">>, attrs = Attrs}]).
+
+enable_stanza(JID, Node) ->
+    enable_stanza(JID, Node, undefined).
+enable_stanza(JID, Node, FormFields) ->
+    enable_stanza(JID, Node, FormFields, ?NS_PUBSUB_PUB_OPTIONS).
+enable_stanza(JID, Node, FormFields, FormType) ->
+    escalus_stanza:iq(<<"set">>, [#xmlel{name = <<"enable">>, attrs = [
+        {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+        {<<"jid">>, JID},
+        {<<"node">>, Node}
+    ], children = maybe_form(FormFields, FormType)}]).
+
+maybe_form(undefined, _FormType) ->
+    [];
+maybe_form(FormFields, FormType) ->
+    [make_form([{<<"FORM_TYPE">>, FormType} | FormFields])].
+
+make_form(Fields) ->
+    #xmlel{name = <<"x">>, attrs = [{<<"xmlns">>, ?NS_XDATA}, {<<"type">>, <<"submit">>}],
+           children = [make_form_field(Name, Value) || {Name, Value} <- Fields]}.
+
+make_form_field(Name, Value) ->
+    #xmlel{name = <<"field">>,
+           attrs = [{<<"var">>, Name}],
+           children = [#xmlel{name = <<"value">>, children = [#xmlcdata{content = Value}]}]}.
+
+
+
+%% ----------------------------------
+%% Other helpers
+%% ----------------------------------
+
+parse_form(#xmlel{name = <<"x">>} = Form) ->
+    parse_form(exml_query:subelements(Form, <<"field">>));
+parse_form(Fields) when is_list(Fields) ->
+    lists:map(
+        fun(Field) ->
+            {exml_query:attr(Field, <<"var">>),
+             exml_query:path(Field, [{element, <<"value">>, cdata}])}
+        end, Fields).
+
+%% @doc Forwards all erlcloud_sns:publish calls to local PID as messages
+start_publish_listener(Config) ->
+    TestCasePid = self(),
+    rpc(meck, new, [ejabberd_router, [no_link, passthrough]]),
+    rpc(meck, expect,
+        [ejabberd_router, route,
+         fun(From, To, Packet) ->
+             case exml_query:subelement(Packet, <<"pubsub">>) of
+                 undefined ->
+                     meck:passthrough([From, To, Packet]);
+                 _ ->
+                     TestCasePid ! #route{from = From, to = To, packet = Packet},
+                     ok
+             end
+         end]),
+    Config.
+
+-spec rpc(M :: atom(), F :: atom(), A :: [term()]) -> term().
+rpc(M, F, A) ->
+    Node = ct:get_config({hosts, mim, node}),
+    Cookie = escalus_ct:get_config(ejabberd_cookie),
+    escalus_ct:rpc_call(Node, M, F, A, 10000, Cookie).
+
+create_room(Room, [Owner | Members]) ->
```

Comment:
Maybe using https://github.com/esl/MongooseIM/blob/master/test.disabled/ejabberd_tests/tests/muc_light_SUITE.erl#L271 would be easier, faster and would require less imports?